typings/index.d.ts

```diff
@@ -0,0 +1,21 @@
+declare module '@walletconnect/web3-provider' {
```

Comment:
I've got a little lost on why we declare these module `declare module '@walletconnect/web3-provider'` here and also in `src/provider/index.ts`. 

What would be the organization/responsabilities of each file?

Reply:
Declaration files are for module imports and for globals
But I can't import our own types (from './types') in there.
So declaration has an incomplete typing of `new (options: WalletConnectOptionsInfura | WalletConnectOptionsRPC)` which I extend is src as `new (options: WalletConnectOptionsInfura | WalletConnectOptionsRPC): WalletConnectProvider`

I also think it's inconvenient.

Reply:
Don't fully understand, but hopefully you know what you are doing there. Resolving, 1<>1 we have a coffee and review it again :)